src/Avalonia.Controls/Flyouts/FlyoutBase.cs

```diff
@@ -0,0 +1,466 @@
+        public void Hide(bool canCancel = true)
+        {
+            if (!IsOpen)
+            {
+                return;
+            }
+
+            if (canCancel)
+            {
+                bool cancel = false;
+
+                var closing = new CancelEventArgs();
+                Closing?.Invoke(this, closing);
+                if (cancel || closing.Cancel)
+                {
+                    return;
+                }
+            }
+
+            IsOpen = false;
+            _popup.IsOpen = false;
+
+            // Ensure this isn't active
+            transientDisposable?.Dispose();
+            transientDisposable = null;
+
+            OnClosed();
+        }
+
+        protected virtual void ShowAtCore(Control placementTarget, bool showAtPointer = false)
+        {
+            if (placementTarget == null)
+                throw new ArgumentNullException("placementTarget cannot be null");
+
+            if (_popup == null)
+            {
+                InitPopup();
+            }
+
+            if (IsOpen)
+            {
+                if (placementTarget == Target)
+                {
+                    return;
+                }
+                else // Close before opening a new one
+                {
+                    Hide(false);
+                }
+            }
+
+            if (_popup.Parent != null && _popup.Parent != placementTarget)
+            {
+                ((ISetLogicalParent)_popup).SetParent(null);
+            }
+
+            if (_popup.PlacementTarget != placementTarget)
+            {
+                _popup.PlacementTarget = Target = placementTarget;
+                ((ISetLogicalParent)_popup).SetParent(placementTarget);
+            }
+
+            if (_popup.Child == null)
+            {
+                _popup.Child = CreatePresenter();
+            }
+
+            OnOpening();
+            PositionPopup(showAtPointer);
+            IsOpen = _popup.IsOpen = true;            
+            OnOpened();
+                        
+            if (ShowMode == FlyoutShowMode.Standard)
+            {
+                // Try and focus content inside Flyout
+                if (_popup.Child.Focusable)
+                {
+                    FocusManager.Instance?.Focus(_popup.Child);
+                }
+                else
+                {
+                    var nextFocus = KeyboardNavigationHandler.GetNext(_popup.Child, NavigationDirection.Next);
+                    if (nextFocus != null)
+                    {
+                        FocusManager.Instance?.Focus(nextFocus);
+                    }
+                }
+            }
+            else if (ShowMode == FlyoutShowMode.TransientWithDismissOnPointerMoveAway)
+            {
+                transientDisposable = InputManager.Instance?.Process.Subscribe(HandleTransientDismiss);
+            }
+        }
+
+        private void HandleTransientDismiss(RawInputEventArgs args)
```

Comment:
This method implementation is hard to understand without comments.
Anyway, it seems to work fine with mouse, but keyboard is not handled in any way (does UWP handle it?).

Reply:
I'll add some comments to help understanding. The `TransientWithDismissOnPointerMoveAway` is mostly a pointer thing, and AFAICT has no keyboard support in UWP (b/c the Popup is never focused). Although b/c `DefaultMenuInteractionHandler` handles most interactions in `MenuFlyout` I did forget keyboard support for `Flyout` so I'll add that in as part of the changes.